internal/sync/lifecycle.go

```diff
@@ -182,6 +184,16 @@ func (l *lifecycleOnce) Stop(f func() error) error {
 	return l.loadError()
 }
 
+// Started returns a channel that will close when the lifecycle starts.
+func (l *lifecycleOnce) Started() <-chan struct{} {
```

Comment:
You probably could do something like `BlockUntilStart` or `WaitUntilStart` instead, and not expose the channel.